evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py

```diff
@@ -84,38 +122,45 @@ def fit(self, X: pd.DataFrame, y: pd.Series = None) -> STLDecomposer:
         self.is_fit = True
         return self
 
-    # @fit_check
     def transform(
         self,
         X: pd.DataFrame,
         y: pd.Series = None,
     ) -> tuple[pd.DataFrame, pd.Series]:
+        if not isinstance(y.index, pd.DatetimeIndex):
+            y = self._set_time_index(X, y)
+
         if not self.is_fit:
             raise ValueError(
                 "STLDecomposer has not been fit yet.  Please fit it and then build the decomposed dataframe.",
             )
 
+        self._check_oos_past(y)
+
         y_in_sample = pd.Series([])
         y_out_of_sample = pd.Series([])
-        if y.index[0] <= self.trend.index[-1] and y.index[0] >= self.trend.index[0]:
-            y_in_sample = self.residual[y.index[0] :]
-        if y.index[-1] > self.trend.index[-1]:
-            truncated_y = y[self.trend.index[-1] :][1:]
 
-            # Determine how many units forward to forecast
-            right_projected_trend = self._project_trend(truncated_y)
+        # For wholly in-sample data, retrieve stored results.
+        if y.index[0] <= self.trend.index[-1] and y.index[0] >= self.trend.index[0]:
+            y_in_sample = self.residual[y.index[0] : y.index[-1]]
 
-            # Reseasonalize
-            projected_seasonal = self._build_seasonal_signal(
-                truncated_y,
-                self.seasonality,
-                self.seasonal_period,
-                self.frequency,
-            )
+        # For out of sample data....
+        if y.index[-1] > self.trend.index[-1]:
+            try:
```

Comment:
Any reason to do try/except? Expecting an error as normal behavior always feels weird to me.

Reply:
It's pythonic :'(